src/Juvix/Compiler/Abstract/Extra/Functions.hs

```diff
@@ -45,14 +45,28 @@ patternCosmos f p = case p of
       args' <- traverse (traverseOf patternArgPattern (patternCosmos f)) args
       pure (PatternConstructorApp (ConstructorApp r args'))
 
+patternArgNameFold :: SimpleFold (Maybe Name) Pattern
+patternArgNameFold f = \case
+  Nothing -> mempty
+  Just n -> Const (getConst (f (PatternVariable n)))
```

Comment:
I probably don't understand but isn't `Const . getConst` the identity? Or are there different `Const`s here?

Reply:
it is needed to get the types right. It differs from identity in the type:
`const . getConst :: Const r a -> Const r b`

Reply:
Thank you